1-foundations/1-types-and-syntax/foundations-s1.ipynb

```diff
@@ -1,11 +1,1580 @@
+    "print(list_int)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a list of strings\n",
+    "list_str = [\"a\",\"b\",\"c\"]\n",
+    "print(list_str)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a mixed list\n",
+    "list_mix = [42,\"Arthur\",False]\n",
+    "print(list_mix)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# test the type of a list\n",
+    "print(type(list_mix))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Access an item in a list**:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Print list and print each item in the list\n",
+    "print('List list_mix:', list_mix, type(list_mix))\n",
+    "print('First item (index 0):', list_mix[0], type(list_mix[0]))\n",
+    "print('Second item (index 1):', list_mix[1], type(list_mix[1]))\n",
+    "print('Third item (index 2):', list_mix[2], type(list_mix[2]))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Access item in list and store in variable\n",
+    "name = list_mix[1]\n",
+    "print('Variable name:', name, type(name))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Accessing items using the index does not modify the list\n",
+    "print('List list_mix:', list_mix, type(list_mix))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Access multiple items in a list:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Re-create list of ints\n",
+    "list_int = [0,1,2,3,4,5,6,7,8,9]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get all items between the item with index 0 \n",
+    "# up until but not including the item with index 3\n",
+    "# 0 ≤ index < 3\n",
+    "print(list_int[0:3])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Index 0 is assumed if the fist number is omitted\n",
+    "# So 0:3 is the same as :3\n",
+    "print(list_int[:3])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 5 ≤ index < 7\n",
+    "print(list_int[5:7])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 8 ≤ index < infinity\n",
+    "print(list_int[8:])"
```

Comment:
I suggest adding a note to emphasize in this point that `[8:]` includes the last element of the list

Reply:
done in: 360840516a909edfe6038351da81f415d8e85b2f